datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/java/de/adorsys/datasafe/simple/adapter/spring/InectionForMinioTest.java

```diff
@@ -0,0 +1,50 @@
+package de.adorsys.datasafe.simple.adapter.spring;
+
+import de.adorsys.datasafe.business.impl.e2e.WithStorageProvider;
+import de.adorsys.datasafe.simple.adapter.api.SimpleDatasafeService;
+import de.adorsys.datasafe.simple.adapter.spring.factory.SpringSimpleDatasafeServiceFactory;
+import de.adorsys.datasafe.simple.adapter.spring.properties.SpringAmazonS3DFSCredentialsProperties;
+import de.adorsys.datasafe.simple.adapter.spring.properties.SpringDFSCredentialProperties;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.stream.Stream;
+
+
+// yes, this is correct. the factory will be instantieated via spring with the filesystem
+// but during runtime we create another factory with minio, right after we got the parameters from
+// the descriptor
+@ActiveProfiles("filesystem")
```

Comment:
There is better solution IMHO, here is patch:
```suggestion
Index: datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/resources/application-minio.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/resources/application-minio.yml	(revision c029676190e935ba348dc84e7415961505b7d459)
+++ datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/resources/application-minio.yml	(date 1562232965000)
@@ -1,10 +1,9 @@
-This file is not used, as the mapped url supplied by docker is known after spring has already done its configuration ....
 datasafe:
   storeconnection:
     amazons3:
-      url: http://localhost
-      accesskey: simpleAccessKey
-      secretkey: simpleSecretKey
+      url: ${MINIO_URL}
+      accesskey: admin
+      secretkey: password
       rootbucket: adorsys
 
 
Index: datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/java/de/adorsys/datasafe/simple/adapter/spring/InectionForMinioTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/java/de/adorsys/datasafe/simple/adapter/spring/InectionForMinioTest.java	(revision c029676190e935ba348dc84e7415961505b7d459)
+++ datasafe-simple-adapter/datasafe-simple-adapter-spring/src/test/java/de/adorsys/datasafe/simple/adapter/spring/InectionForMinioTest.java	(date 1562233019000)
@@ -3,12 +3,12 @@
 import de.adorsys.datasafe.business.impl.e2e.WithStorageProvider;
 import de.adorsys.datasafe.simple.adapter.api.SimpleDatasafeService;
 import de.adorsys.datasafe.simple.adapter.spring.factory.SpringSimpleDatasafeServiceFactory;
-import de.adorsys.datasafe.simple.adapter.spring.properties.SpringAmazonS3DFSCredentialsProperties;
 import de.adorsys.datasafe.simple.adapter.spring.properties.SpringDFSCredentialProperties;
 import lombok.extern.slf4j.Slf4j;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.test.context.ActiveProfiles;
 
 import java.util.stream.Stream;
@@ -17,32 +17,31 @@
 // yes, this is correct. the factory will be instantieated via spring with the filesystem
 // but during runtime we create another factory with minio, right after we got the parameters from
 // the descriptor
-@ActiveProfiles("filesystem")
+//
 
 @Slf4j
+@ActiveProfiles("minio")
 public class InectionForMinioTest extends InjectionTest {
 
+    @Autowired
+    private SpringDFSCredentialProperties dfsCredentialProperties;
+
+    @BeforeAll
+    static void startMinio() {
+        minio().getStorageService().get();
+        System.setProperty("MINIO_URL",  minio().getMappedUrl());
+    }
+
     private static Stream<StorageDescriptor> minioonly() {
-        return allStorages().filter(storage -> storage.getName().equals(StorageDescriptorName.MINIO));
+        return Stream.of(minio());
     }
 
     @ParameterizedTest
     @MethodSource("minioonly")
     public void plainService(WithStorageProvider.StorageDescriptor descriptor) {
-
         log.info("descriptor is " + descriptor.getName());
-        descriptor.getStorageService().get();
-        SpringAmazonS3DFSCredentialsProperties s3properties = new SpringAmazonS3DFSCredentialsProperties();
-        s3properties.setAccesskey(descriptor.getAccessKey());
-        s3properties.setSecretkey(descriptor.getSecretKey());
-        s3properties.setRegion(descriptor.getRegion());
-        s3properties.setUrl(descriptor.getMappedUrl());
-        s3properties.setRootbucket("target");
-        SpringDFSCredentialProperties properties = new SpringDFSCredentialProperties();
-        properties.setAmazons3(s3properties);
-        log.info("properties for miniotest created by test look:" + properties);
         DatasafeSpringBeans s = new DatasafeSpringBeans();
-        SpringSimpleDatasafeServiceFactory springSimpleDatasafeServiceFactory = s.simpleDatasafeServiceFactory(properties);
+        SpringSimpleDatasafeServiceFactory springSimpleDatasafeServiceFactory = s.simpleDatasafeServiceFactory(dfsCredentialProperties);
 
         SimpleDatasafeService service = springSimpleDatasafeServiceFactory.getSimpleDataSafeServiceWithSubdir("subdir");
         testCreateUser(service);
```

Reply:
yes, great. works fine. 
Drawback is stupid code coverage which thinks my SpringConfigBean is not tested. I will ignore it in codecoverage submodule as soon as I see my bean with read untested.